backend/src/libs/interfaces/service.interface.ts

```diff
@@ -0,0 +1,11 @@
+interface IService<T = unknown> {
+  find(): Promise<T>;
+  findAll(): Promise<{
+    items: T[];
+  }>;
+  create(payload: unknown): Promise<T>;
+  update(): Promise<T>;
+  delete(): Promise<boolean>;
+}
```

Comment:
Maybe it makes sense to make all these methods optional and in IRepository too? it's not flexible and if it's a little strange that user service implements this interface while auth service not

Reply:
User should use `Omit` in this case

Reply:
`class AuthService implements Omit<IService, 'find' | 'findAll' | 'create' | 'update' | 'delete'>`?

Reply:
right

Reply:
then do this for AuthService for now it doesn't implement IService